File: pa1.c
/* Taylor Coury, Melinda Grad
 * Automata PA 1 
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <limits.h>

// TODO: Function prototypes


int main(int argc, char *argv[]){

	if(argc != 2){
		printf("Program needs one argument <filename>\n");
		exit(1);
	}

	// Read the file containing the DFA simulation specifications
	FILE *infile = fopen(argv[1], "r");
	if(infile ==NULL){
		fprintf(stderr, "Error: Could not open file %s for reading\n", argv[1]);
		exit(1);
	}

	// Read first line containing number of states
	int num_states = 0;
	fscanf(infile, "%d", &num_states);
	printf("Number of states: %d\n", num_states);

	//Get the alphabet
	char *alpha = malloc(100*sizeof(char));
	fscanf(infile, "%s", alpha);
	//printf("%c\n", alpha[0]);

	//Get the number of characters in the aplhabet
	int num_alpha = strlen(alpha);
	printf("num_alpha = %d\n", num_alpha);
	
	int num_transitions = num_alpha * num_states;
	int transition_array[num_alpha][num_states];

	for(int i = 0; i< num_transitions; i++){
		int curr_state = 0;
		int next_state = 0;
		char input;

		//Get the transition
		fscanf(infile, "%d '%c' %d", &curr_state, &input, &next_state);
		
		// Checking which element of the alphabet
		for(int j =0; j< num_alpha; j++){
			printf("alpha[j] = %c %c\n", alpha[j], input);
			int ret = strcmp(input, alpha[i]);
			if(ret ==0){
				// Put the input into the array

				transition_array[j][curr_state-1] = next_state;
				printf(" element: %d", transition_array[j][curr_state-1]);
				break;
			}

	///TAke in the accept tates string, tokenize by space, update counter, change each token to
	// and put into an int array to hold all the accept states.
	//TODO: 

	//TODO: Read in (for loop) the strings

	//TODO:

	return 0;
}

File: pa1.java
import java.io.*;
import java.util.Scanner;


public class pa1{

    public static void main (String[] args){
    /*
        Scanner scanner = new Scanner(System.in);
        String file_name = scanner.next();
        scanner.close();
    */

        //File file = new File(file_name);
        File file = new File(args[0]);
        
        Scanner file_sc = null;
        try{
            file_sc = new Scanner(file);
        }
        catch(FileNotFoundException e){
            System.out.println(e);
        }
        
        /*TODO:
        while(file_sc.hasNextLine()){
            String s = file_sc.nextLine();
            System.out.println(s);
        }*/
        // Get the number of states
        String s = file_sc.nextLine();
        int num_states = Integer.parseInt(s);
        System.out.println(num_states);

        //Get the alphabet
        String alphabet = file_sc.nextLine();
        System.out.println(alphabet);

        //Get number transitions
        int num_alpha = alphabet.length();
        //System.out.println(num_alpha);
        int num_transitions = num_alpha * num_states;

        //Array to hold transitions
        int[][] transition_array = new int[num_alpha][num_states]; 
        //Loop to populate the array of transitions
        for(int i=0; i<num_transitions; i++){
            //Read in the transition
            int curr_state = Integer.parseInt(file_sc.next());
            String input = file_sc.next();
            int next_state = Integer.parseInt(file_sc.next());

            //Get rid of ''
            String clean_input = input.replace("'", "");
            
            for(int j=0; j< num_alpha; j++){
                if(clean_input.compareTo(Character.toString(alphabet.charAt(j)))==0){
                    transition_array[j][curr_state-1] = next_state;
                    break;
                }
            }
        }
        /*Prints the array
        for(int i=0; i< num_alpha; i++){
            System.out.println("\n");
            for(int j=0; j<num_states; j++){
                System.out.println(transition_array[i][j]);
            }
        }*/

        int start_state = Integer.parseInt(file_sc.next());
        System.out.println(start_state);

        //TODO: Get the set of accept states
        String temp = file_sc.nextLine();
        String accept_temp = file_sc.nextLine();
        System.out.println(accept_temp);
        String[] accept_states = accept_temp.split(" ");
        int num_accept = accept_states.length;
        for(int i = 0; i < num_accept; i++){
            System.out.println(accept_states[i]);
        }
        
        //TODO: Get the input strings to simulate then check against the array
        /*while(file_sc.hasNextLine()){
            String input = file_sc.nextLine();
            int length = input.length();
            int current = start_state;
            for(int i = 0; i < length; i++){
                
            }
        }
        */


        file_sc.close();

    }
}
